core/frontend/src/extension/ExtensionRuntime.ts

```diff
@@ -0,0 +1,103 @@
+import { ACSDisplayOptions, ACSType } from "../AuxCoordSys";
+import { CoordSystem } from "../CoordSystem";
+import { LocateAction, LocateFilterStatus, LocateResponse, SnapStatus } from "../ElementLocateManager";
+import { FlashMode } from "../FlashSettings";
+import { FrontendLoggerCategory } from "../FrontendLoggerCategory";
+import { HitDetailType, HitGeomType, HitParentGeomType, HitPriority, HitSource, SnapHeat, SnapMode } from "../HitDetail";
+import { ActivityMessageEndReason, MessageBoxIconType, MessageBoxType, MessageBoxValue, NotifyMessageDetails, OutputMessageAlert, OutputMessagePriority, OutputMessageType } from "../NotificationManager";
+import { SelectionSetEventType } from "../SelectionSet";
+import { StandardViewId } from "../StandardView";
+import { ViewRect } from "../ViewRect";
+import { ViewStatus } from "../ViewStatus";
+import { DepthPointSource, ViewUndoEvent } from "../Viewport";
+import { GraphicType } from "../render/GraphicBuilder";
+import { Pixel } from "../render/Pixel";
+import { UniformType, VaryingType } from "../render/ScreenSpaceEffectBuilder";
+import { BingElevationProvider, readElementGraphics, TileBoundingBoxes, TileGraphicType, TileLoadPriority, TileLoadStatus, TileTreeLoadStatus, TileVisibility } from "../tile/internal";
+import { ClipEventType } from "../tools/ClipViewTool";
+import { PrimitiveTool } from "../tools/PrimitiveTool";
+import { SelectionMethod, SelectionMode, SelectionProcessing } from "../tools/SelectTool";
+import { BeButton, BeButtonEvent, BeModifierKeys, CoordinateLockOverrides, CoordSource, EventHandled, InputCollector, InputSource, InteractiveTool, KeyinParseError, ParseAndRunResult, Tool } from "../tools/Tool";
+import { ManipulatorToolEvent, StartOrResume } from "../tools/ToolAdmin";
+import { ToolAssistance, ToolAssistanceImage, ToolAssistanceInputMethod } from "../tools/ToolAssistance";
+import { ViewTool } from "../tools/ViewTool";
+import { EditManipulator } from "../tools/EditManipulator";
+import { EmphasizeElements } from "../EmphasizeElements";
+import { FeatureSymbology } from "../render/FeatureSymbology";
+import { GraphicBranch } from "../render/GraphicBranch";
+import { canvasToImageBuffer, canvasToResizedCanvasWithBars, extractImageSourceDimensions, getCompressedJpegFromCanvas, getImageSourceFormatForMimeType, getImageSourceMimeType,
+  imageBufferToBase64EncodedPng, imageBufferToCanvas, imageBufferToPngDataUrl, imageElementFromImageSource, imageElementFromUrl } from "../ImageUtil";
+import { queryTerrainElevationOffset } from "../ViewGlobalLocation";
+import { BingLocationProvider } from "../BingLocation";
+import { FrustumAnimator } from "../FrustumAnimator";
+import { GlobeAnimator } from "../GlobeAnimator";
+import { MarginPercent } from "../MarginPercent";
+import { Cluster, Marker } from "../Marker";
+
+const globalSymbol = Symbol.for("itwin.core.frontend.globals");
+if ((globalThis as any)[globalSymbol])
```

Comment:
@wgoehrig does it make sense to add this to other pkgs?

Reply:
Agree, I think it does for at the very least core-backend since you can never have multiple copies of that. However, not sure about the other ones in core. Their requirement to have a single one is driven by the core-frontend and core-backend pkgs.

Now some of the UI packages are another set that it may be helpful to do it for, mainly the framework ones.

Reply:
@aruniverse added this to the backlog to at least tackle the core-backend.